src/app/views/river-detail/river-creator.vue

```diff
@@ -0,0 +1,284 @@
+            <cv-select-option
+              v-for="item in $options.reachClasses"
+              :key="item"
+              :value="item"
+            >
+              {{ item }}
+            </cv-select-option>
+          </cv-select>
+          <label for="Content Editor" class="bx--label">Description</label>
+          <content-editor
+              :content="!reach.description ? ' ' : reach.description"
+              show-control-bar
+              @content:updated="reach.description = $event" />
+          <div class="mb-spacing-md mt-spacing-md"/>
+          <cv-link
+            @click="additionalFieldsVisible = !additionalFieldsVisible"
+            @keydown.enter="additionalFieldsVisible = !additionalFieldsVisible"
+          >
+            {{
+              `${additionalFieldsVisible ? "Hide" : "Show"} Additional Fields`
+            }}
+          </cv-link>
+          <div class="mb-spacing-lg"/>
+          <template v-if="additionalFieldsVisible">
+            <cv-text-input
+              v-model="reach.agency"
+              label="Agency"
+              class="mb-spacing-md"
+            />
+            <cv-text-input
+              v-model="reach.county"
+              label="County"
+              class="mb-spacing-md"
+            />
+            <cv-text-input
+              v-model="reach.zipcode"
+              label="Zipcode"
+              class="mb-spacing-md"
+            />
+            <cv-text-input
+              v-model="reach.length"
+              label="Length"
+              class="mb-spacing-md"
+            />
+            <cv-text-input
+              v-model="reach.permiturl"
+              label="Permit URL"
+              class="mb-spacing-md"
+            />
+            <cv-text-input
+              v-model="reach.permitinfo"
+              label="Permit Info"
+              class="mb-spacing-md"
+            />
+            <cv-text-input
+              v-model="reach.avggradient"
+              label="Average Gradient"
+              class="mb-spacing-md"
+            />
+            <cv-text-input v-model="reach.maxgradient" label="Max Gradient" />
+            <div class="mb-spacing-lg"/>
+          </template>
+          <cv-button-set>
+          <cv-button
+            :disabled="createLoading"
+            @click.exact="handleSubmit"
+            @keydown.enter="handleSubmit"
+          >
+            Submit
+          </cv-button>
+          </cv-button-set>
+        </div>
+        
+      </div>
+    </div>
+  </div>
+</template>
+<script>
+import { createReach } from "@/app/services";
+import turfLength from '@turf/length'
+import GeometryEditor from "@/app/views/river-detail/components/geometry-edit-modal/components/geometry-editor.vue"
+import ContentEditor from '@/app/global/components/content-editor/content-editor.vue'
+export default {
+  name: "river-creator",
+  components: {
+    GeometryEditor,
+    ContentEditor
+  },
+  /* eslint-disable-next-line max-len */
+  reachClasses: [
```

Comment:
I created a mixin for this a month or so ago: https://github.com/AmericanWhitewater/wh2o-vue/blob/development/src/app/global/mixins/reach-classes.js